OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -296,6 +296,14 @@ public WDist GetMaximumRangeVersusTarget(Target target)
 		// Enumerates all armaments, that this actor possesses, that can be used against Target t
 		public IEnumerable<Armament> ChooseArmamentsForTarget(Target t, bool forceAttack)
 		{
+			// If the target is frozen, make sure it is still valid
+			if (t.Type == TargetType.FrozenActor)
+			{
+				var frozenLayer = self.Owner.PlayerActor.TraitOrDefault<FrozenActorLayer>();
+				if (frozenLayer == null || frozenLayer.FromID(t.FrozenActor.ID) == null)
```

Comment:
You should be able to use `t.FrozenActor` directly here, instead of querying it again from the layer?